src/base/Pie.i.hh

```diff
@@ -14,65 +13,80 @@ namespace celeritas
 /*!
  * Construct with a particular range.
  */
-template<class T>
-CELER_FUNCTION PieSlice<T>::PieSlice(size_type start, size_type stop)
+template<class T, class S>
+CELER_FUNCTION PieSlice<T, S>::PieSlice(size_type start, size_type stop)
     : start_(start), stop_(stop)
 {
     CELER_EXPECT(start_ <= stop_);
 }
 
+//---------------------------------------------------------------------------//
+/*!
+ * Construct with a particular range.
+ */
+template<class T, class S>
+CELER_FUNCTION auto PieSlice<T, S>::operator[](size_type i) const -> value_type
+{
+    CELER_EXPECT(i < stop_);
```

Comment:
```suggestion
    CELER_EXPECT(start_ <= i && i < stop_);
```

Reply:
Actually it should be `i < this->size()` (i.e. `start_ + i < stop_`). Good catch!